src/uniprotkb/components/entry/__tests__/SimilarProteins.spec.tsx

```diff
@@ -1,37 +1,39 @@
-import renderWithRouter from '../../../../shared/__test-helpers__/RenderWithRouter';
-import SimilarProteins from '../SimilarProteins';
+import { act } from 'react-dom/test-utils';
+import { fireEvent } from '@testing-library/react';
+import renderWithRedux from '../../../../shared/__test-helpers__/RenderWithRedux';
+import SimilarProteins from '../similar-proteins/SimilarProteins';
 import similarProteinsData from './__mocks__/similarProteinsData.json';
+import accessionsData from './__mocks__/accessionsData.json';
 
-jest.mock('../../../../shared/hooks/useDataApi', () => jest.fn());
-import useDataApi from '../../../../shared/hooks/useDataApi';
-import { fireEvent } from '@testing-library/react';
+var axios = require('axios');
+var MockAdapter = require('axios-mock-adapter');
 
-const dataMock = {
-  loading: false,
-  data: similarProteinsData,
-};
+var axiosMock = new MockAdapter(axios);
+axiosMock.onGet(/\/uniref\/search/).reply(200, similarProteinsData);
+axiosMock.onGet(/\/uniprotkb\/accessions/).reply(200, accessionsData);
 
-const getRendered = () =>
-  renderWithRouter(
-    <SimilarProteins
-      primaryAccession="P05067"
-      isoforms={{ isoforms: ['P05067-4'] }}
-    />
-  );
-describe('SimilarProteins tests', () => {
-  beforeEach(() => {
-    useDataApi.mockImplementation(() => dataMock);
+const getRendered = async () => {
+  let rendered;
+  await act(async () => {
+    rendered = renderWithRedux(
+      <SimilarProteins
+        primaryAccession="P05067"
+        isoforms={{ isoforms: ['P05067-4'] }}
+      />
+    );
   });
+  return rendered;
+};
 
+describe('SimilarProteins tests', () => {
   it('should call useDataApi and render', async () => {
-    const { findByText } = getRendered();
-    expect(useDataApi).toHaveBeenCalled();
-    expect(await findByText(/0FGN2/)).toBeTruthy();
+    const { findAllByText } = await getRendered();
+    expect(await findAllByText(/0FGN2/)).toBeTruthy();
   });
 
   it('should change tabs', async () => {
-    const { getByText, findByText } = getRendered();
-    fireEvent.click(getByText('50% identity'));
+    const { findByText } = await getRendered();
+    fireEvent.click(await findByText(/50% identity/));
```

Comment:
```suggestion
    fireEvent.click(await screen.findByRole('button', { name: /50% identity/ }));
```
See https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen and https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-byrole-most-of-the-time (Especially if we're going to click it, it would be good to assert that this is a button by using this query)